internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```diff
@@ -111,4 +110,93 @@ NLBDNSAlias:
+{{- end}}
+{{- if eq .NLB.Listener.Protocol "TLS" }}
+NLBCertManagerAction:
+  Metadata:
+    'aws:copilot:description': "Request and validate the certificate for your Network Load Balancer, as well as adding A-records for the aliases."
+  Type: Custom::NLBCertManagerFunction
+  Condition: HasAssociatedDomain
+  Properties:
+    ServiceToken: !Ref NLBCertManagerFunction
+    LoadBalancerDNS: !GetAtt PublicNetworkLoadBalancer.DNSName
+    LoadBalancerHostedZoneID: !GetAtt PublicNetworkLoadBalancer.CanonicalHostedZoneID
+    EnvHostedZoneId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-HostedZone"
+    EnvName: !Ref EnvName
+    AppName: !Ref AppName
+    ServiceName: !Ref WorkloadName
+    RootDNSRole: {{ .AppDNSDelegationRole }}
+    DomainName:  {{ .AppDNSName }}
+    Aliases: {{ .NLB.Listener.Aliases }}
+
+NLBCertManagerFunction:
+  Type: AWS::Lambda::Function
+  Condition: HasAssociatedDomain
+  Properties:
+    Code:
+      ZipFile: |
+        {{ .NLBCertManagerFunctionLambda }}
+    Handler: "index.handler"
+    Timeout: 900
+    MemorySize: 512
+    Role: !GetAtt 'NLBCertManagerRole.Arn'
+    Runtime: nodejs12.x
+
+NLBCertManagerRole:
+  Type: AWS::IAM::Role
+  Properties:
+    AssumeRolePolicyDocument:
+      Version: 2012-10-17
+      Statement:
+        -
+          Effect: Allow
+          Principal:
+            Service:
+              - lambda.amazonaws.com
+          Action:
+            - sts:AssumeRole
+    Path: /
+    Policies:
+      - PolicyName: "NLBCertManagerPolicyWrite"
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Sid: EnvHostedZoneUpdateAndWait
+              Effect: Allow
+              Action: route53:ChangeResourceRecordSets
+              Resource:
+                !Sub
+                  - arn:${AWS::Partition}:route53:::hostedzone/${EnvHostedZone}
+                  - EnvHostedZone: Fn::ImportValue
+                        !Sub "${AppName}-${EnvName}-HostedZone"
+            - Sid: EnvHostedZoneRead
+              Effect: Allow
+              Action:
+                - route53:ListResourceRecordSets
+                - route53:GetChange
+              Resource: "*"
+            - Sid: ServiceCertificateDelete
+              Effect: Allow
+              Action: acm:DeleteCertificate
+              Resource: "*"
+              Condition:
+                StringEquals:
+                  'aws:ResourceTag/copilot-application': !Sub '${AppName}'
+                  'aws:ResourceTag/copilot-environment': !Sub '${EnvName}'
+                  'aws:ResourceTag/copilot-service': !Sub '${WorkloadName}'
+            - Sid: TaggedResourcesRead
+              Effect: Allow
+              Action: tag:GetResources
+              Resource: "*"
+            - Sid: CertificateCreateAndRead
+              Effect: Allow
+              Action:
+                - acm:DescribeCertificate
+                - acm:RequestCertificate
+                - acm:AddTagsToCertificate
```

Comment:
The permissions look good! Do you know by any chance if for these two actions doing a condition on `aws:RequestTag/${TagKey} ` would narrow it down further?
https://docs.aws.amazon.com/service-authorization/latest/reference/list_awscertificatemanager.html#awscertificatemanager-policy-keys

Reply:
yes! updated!